packages/mdc-chips/chip/foundation.ts

```diff
@@ -94,19 +94,24 @@ export class MDCChipFoundation extends MDCFoundation<MDCChipAdapter> {
   }
 
   getDimensions(): ClientRect {
-    const rootRect = this.adapter_.getRootBoundingClientRect();
-    const checkmarkRect = this.adapter_.getCheckmarkBoundingClientRect();
+    const getRootRect = () => this.adapter_.getRootBoundingClientRect();
+    const getCheckmarkRect = () => this.adapter_.getCheckmarkBoundingClientRect();
 
     // When a chip has a checkmark and not a leading icon, the bounding rect changes in size depending on the current
     // size of the checkmark.
-    if (!this.adapter_.hasLeadingIcon() && checkmarkRect !== null) {
-      // The checkmark's width is initially set to 0, so use the checkmark's height as a proxy since the checkmark
-      // should always be square.
-      const width = rootRect.width + checkmarkRect.height;
-      return {...rootRect, width};
-    } else {
-      return rootRect;
+    if (!this.adapter_.hasLeadingIcon()) {
+      const checkmarkRect = getCheckmarkRect();
+      if (checkmarkRect) {
+        const rootRect = getRootRect();
+        const height = rootRect.height;
+        // The checkmark's width is initially set to 0, so use the checkmark's height as a proxy since the checkmark
+        // should always be square.
+        const width = rootRect.width + checkmarkRect.height;
+        return {...rootRect, width, height};
+      }
```

Comment:
There's a missing return in the case of checkmarkRect being falsy. However, tests still seem to be passing. Do we still need the falsy check for `checkmarkRect`?

Reply:
The missing `return` statement is logically equivalent to the original syntax; I just split a single `if(...)` statement into two for better readability:

https://github.com/material-components/material-components-web/blob/70672f12ff19591c53fef6876c059c854f9cb816/packages/mdc-chips/chip/foundation.js#L116-L129

`checkmarkRect` can be `null` because `adapter.getCheckmarkBoundingClientRect()` may return `null`, so I think we probably need to keep the falsy check (unless I'm misunderstanding).

Reply:
I GET IT NOW. My bad... It's been a long day.

Reply:
I hear that 😴